plugins/org.eclipse.xtext.xtext.generator/src/org/eclipse/xtext/xtext/generator/ui/contentAssist/ContentAssistFragment2.xtend

```diff
@@ -0,0 +1,330 @@
+import org.eclipse.xtext.xtext.generator.model.TypeReference
+
+import static extension org.eclipse.xtext.GrammarUtil.*
+import static extension org.eclipse.xtext.xtext.generator.util.GrammarUtil2.*
+
+/**
+ * Contributes the 'Abstract...ProposalProvider' and '...ProposalProvider' stub,
+ *  the latter either in Xtend or Java language.
+ * 
+ * @author Christian Schneider - Initial contribution and API
+ */
+class ContentAssistFragment2 extends AbstractGeneratorFragment2 {
+
+	@Inject
+	extension XtextGeneratorNaming
+	
+	@Inject
+	extension CodeConfig
+	
+	@Inject
+	FileAccessFactory fileAccessFactory
+
+	@Accessors
+	boolean generateStub = true;
+
+	@Accessors
+	boolean inheritImplementation = true
+
+	def protected TypeReference getProposalProviderClass(Grammar g) {
+		return new TypeReference(
+			g.eclipsePluginBasePackage + ".contentassist." + g.simpleName + "ProposalProvider"
+		)
+	}
+
+	def protected TypeReference getGenProposalProviderClass(Grammar g) {
+		return new TypeReference(
+			g.eclipsePluginBasePackage + ".contentassist.Abstract" + g.simpleName + "ProposalProvider"
+		)
+	}
+
+	def protected TypeReference getGenProposalProviderSuperClass(Grammar g) {
+		val superGrammar = g.nonTerminalsSuperGrammar
+		if(inheritImplementation && superGrammar != null)
+			superGrammar.proposalProviderClass
+		else new TypeReference(
+			"org.eclipse.xtext.ui.editor.contentassist.AbstractJavaBasedContentProposalProvider"
+		)
+	}
+
+	/**
+	 * Extra getter facilitates customization by overriding. 
+	 */
+	def protected TypeReference getDefaultEObjectLabelProviderSuperClass() {
+		new TypeReference("org.eclipse.xtext.ui.label.DefaultEObjectLabelProvider")
+	}
+
+
+	override generate() {
+		val chosenClass = 
+			if (generateStub) grammar.getProposalProviderClass else grammar.getGenProposalProviderClass;
+		
+		if (projectConfig.eclipsePluginManifest != null) {
+			projectConfig.eclipsePluginManifest.requiredBundles += "org.eclipse.xtext.ui"
+		}
+
+		new GuiceModuleAccess.BindingFactory()
+				.addTypeToType(
+					new TypeReference("org.eclipse.xtext.ui.editor.contentassist.IContentProposalProvider"),
+					chosenClass
+				).contributeTo(language.eclipsePluginGenModule);
+
+		var generatedCode = false
+
+		if (projectConfig.eclipsePluginSrcGen !== null) {
+			// generate the 'Abstract...ProposalProvider'
+			generateGenJavaProposalProvider
+			generatedCode = true
+		}
+
+		if (generateStub && projectConfig.eclipsePluginSrc != null) {
+			if (preferXtendStubs) {
+				generateXtendProposalProviderStub
+
+				if (projectConfig.eclipsePluginManifest != null) {
+					projectConfig.eclipsePluginManifest.requiredBundles += "org.eclipse.xtext.xbase.lib"
+				}
+			} else {
+				generateJavaProposalProviderStub
+			}
+
+			generatedCode = true
+		}
+
+		if (generatedCode && projectConfig.eclipsePluginManifest != null) {
+			projectConfig.eclipsePluginManifest.exportedPackages += grammar.proposalProviderClass.packageName
```

Comment:
You can skip the "generatedCode" check. A project with a manifest but neither src nor srcGen is broken.